libcrux-ml-kem/src/sampling.rs

```diff
@@ -46,37 +43,39 @@ use crate::{
 ///
 /// The NIST FIPS 203 standard can be found at
 /// <https://csrc.nist.gov/pubs/fips/203/ipd>.
-fn sample_from_uniform_distribution_next<const K: usize, const N: usize>(
+#[inline(always)]
+fn sample_from_uniform_distribution_next<Vector: Operations, const K: usize, const N: usize>(
     randomness: [[u8; N]; K],
     sampled_coefficients: &mut [usize; K],
     out: &mut [[i16; 256]; K],
 ) -> bool {
-    let mut done = true;
+    // Would be great to trigger auto-vectorization or at least loop unrolling here
```

Comment:
In this case, this would need to the platform dependent crates. Do you want to do that here?

Reply:
Right, but for the compiler to do it with all the hardware instructions, it needs to move. Leaving this for later.